core/src/main/scala/chisel3/Module.scala

```diff
@@ -140,6 +140,17 @@ object Module extends SourceInfoDoc {
 
 }
 
+/** Allowed values for the types of Module.reset */
+sealed trait ModuleResetType
+/* The default reset type. This is Uninferred, unless it is the top Module, in which case it is Bool */
+case object ModuleResetTypeDefault extends ModuleResetType
+/* Explicitly Uninferred Reset, even if this is the top Module */
+case object ModuleResetTypeUninferred extends ModuleResetType
+/* Explicitly Bool (Synchronous) Reset */
+case object ModuleResetTypeBool extends ModuleResetType
+/* Explicitly Asynchronous Reset */
+case object ModuleResetTypeAsync extends ModuleResetType
```

Comment:
Style nit: I'm not a big fan of the repeated names. What about:

```scala
object ModuleResetType { // This could just be "ResetType"
  sealed trait Type
  case object Default extends Type
  case object Uninferred extends Type
  case object Synchronous extends Type
  case object Asynchronous extends Type
}
```

This still gets the full naming, but done via a package, e.g., `ModuleReset.Asynchronous`.

Reply:
agree, and I'll ask if you'd accept one step further... 

```
object Module {
  object ResetType {
    sealed trait Type
    case object Default extends Type
    ...
  }
}
```

thoughts?

Reply:
i have implemented my version